### PR TITLE
Use README as crate docs

### PR DESCRIPTION
**Pull request by newpavlov**
Moves all crate-level docs to appropriate READMEs. Additionally, adds some cleanups made possible by migration to `hybrid-array` and `hex-literal v0.4`.

### CODE REVIEW

File: blake2/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ hex-literal = "0.4"
 default = ["std"]
 std = ["digest/std"]
 reset = [] # Enable reset functionality
-simd = []
-simd_opt = ["simd"]
-simd_asm = ["simd_opt"]
+#simd = []
+#simd_opt = ["simd"]
+#simd_asm = ["simd_opt"]
```
Comment (newpavlov) [inline blake2/Cargo.toml:28]: We probably should finish #228 before publishing blake2 v0.11.

Comment (tarcieri) [reply blake2/Cargo.toml:28]: Yeah, definitely been meaning to do it. Would be great to get it over the finish line.

File: sha2/README.md [hunk 2]
```diff
@@ -7,12 +7,52 @@
 [![Project Chat][chat-image]][chat-link]
 [![Build Status][build-image]][build-link]
 
-Pure Rust implementation of the [SHA-2 hash function family][1]
-including SHA-224, SHA-256, SHA-384, and SHA-512.
+Pure Rust implementation of the [SHA-2] cryptographic hash algorithms.
 
-[Documentation][docs-link]
+There are 6 standard algorithms specified in the SHA-2 standard: 
+[`Sha224`], [`Sha256`], [`Sha512_224`], [`Sha512_256`], [`Sha384`], and [`Sha512`].
```
Comment (tarcieri) [inline sha2/README.md]: These are dangling links when rendered on GitHub

File: sha2/README.md [hunk 3]
```diff
@@ -7,12 +7,52 @@
...
 
-<img src="https://raw.githubusercontent.com/RustCrypto/meta/master/img/hashes/sha2.png" width="480px">
```
Comment (tarcieri) [inline sha2/README.md:15]: Aww 😄

File: skein/README.md [hunk 4]
```diff
@@ -7,9 +7,32 @@
...
 
-Pure Rust implementation of the [Skein] family of cryptographic hash algorithms.
+Implementation of the [Skein] family of cryptographic hash algorithms.
 
-[Documentation][docs-link]
+There are 3 standard versions of the Skein hash function:
+
+* [`Skein256`]
+* [`Skein512`]
+* [`Skein1024`]
```
Comment (tarcieri) [inline skein/README.md]: Some more dangling links when rendered on GitHub

File: sha2/README.md (hunk 2 above)
Comment (newpavlov) [reply sha2/README.md]: Hm... I wonder how it's better to handle those. Simply remove linking? Adding explicit links would be... annoying.

File: sha2/README.md (hunk 3 above)
Comment (newpavlov) [reply sha2/README.md:15]: I think most users of the crate (and thus readers of the doc) would not care for this pretty picture. :)

Comment (newpavlov) [reply sha2/README.md:15]: I can add it after examples if you want.

Comment (tarcieri) [reply sha2/README.md:15]: Whatever

**Review by tarcieri — APPROVED** [review]

**Pull request merged**